src/embedded/embeddedManager.ts

```diff
@@ -141,4 +154,62 @@ export class EmbeddedManager {
   public getUpdateHandlers(): Array<EmbeddedMessageUpdateHandler> {
     return this.updateListeners;
   }
+
+  handleEmbeddedClick(
+    message: any,
+    buttonIdentifier: string | null,
+    clickedUrl: string | null
+  ) {
+    if (clickedUrl && clickedUrl.trim() !== '') {
+      let actionType: string;
+      let actionName: string;
+
+      if (clickedUrl.startsWith(URL_SCHEME_ACTION)) {
+        actionType = URL_SCHEME_ACTION;
+        actionName = clickedUrl.replace(URL_SCHEME_ACTION, '');
+      } else if (clickedUrl.startsWith(URL_SCHEME_ITBL)) {
+        actionType = URL_SCHEME_ITBL;
+        actionName = clickedUrl.replace(URL_SCHEME_ITBL, '');
+      } else {
+        actionType = URL_SCHEME_OPEN;
+        actionName = clickedUrl.replace(URL_SCHEME_OPEN, '');
+      }
+
+      const iterableAction: IterableAction = {
+        type: actionType,
+        data: actionName
+      };
+
+      IterableActionRunner.executeAction(
+        null,
+        iterableAction,
+        IterableActionSource.EMBEDDED
+      );
+    }
+  }
+
+  trackEmbeddedClick(
+    message: any,
+    buttonIdentifier: string | '',
+    clickedUrl: string | ''
+  ) {
+    const payload = {
+      messageId: message?.metadata?.messageId,
+      campaignId: message?.metadata?.campaignId
+    };
+
+    const emailOrUserId =
```

Comment:
We shouldn't be tracking the click if there is nothing found in local storage. Shouldn't happen, but we should account for it. And ideally, `email` and `userId` should be fields passed in to this method so that the search for local storage is handled outside the scope of this method.

Reply:
I've implemented a check for both the `email` and `userId` in the Android SDK, following the same approach used in other methods. During API calls, the `email` and `userId` are added to the request, ensuring consistency across our SDK architectures.